code/datums/status_effects/buffs.dm

```diff
@@ -61,108 +54,32 @@
 /datum/status_effect/blooddrunk/on_apply()
 	. = ..()
 	if(.)
-		owner.maxHealth *= 10
-		owner.bruteloss *= 10
-		owner.fireloss *= 10
 		if(ishuman(owner))
 			var/mob/living/carbon/human/H = owner
 			for(var/X in H.bodyparts)
 				var/obj/item/organ/external/BP = X
-				BP.max_damage *= 10
-				BP.min_broken_damage *= 10
-				BP.brute_dam *= 10
-				BP.burn_dam *= 10
-		owner.toxloss *= 10
-		owner.oxyloss *= 10
-		owner.cloneloss *= 10
-		owner.staminaloss *= 10
-		owner.updatehealth()
-		last_health = owner.health
-		last_bruteloss = owner.getBruteLoss()
-		last_fireloss = owner.getFireLoss()
-		last_toxloss = owner.getToxLoss()
-		last_oxyloss = owner.getOxyLoss()
-		last_cloneloss = owner.getCloneLoss()
-		last_staminaloss = owner.getStaminaLoss()
+				BP.brute_mod *= 0.1
+				BP.burn_mod *= 0.1
+			H.dna.species.tox_mod *= 0.1
+			H.dna.species.oxy_mod *= 0.1
+			H.dna.species.clone_mod *= 0.1
+			H.dna.species.stamina_mod *= 0.1
 		add_attack_logs(owner, owner, "gained blood-drunk stun immunity")
 		var/status = CANSTUN | CANWEAKEN | CANPARALYSE | IGNORESLOWDOWN
 		owner.status_flags &= ~status
 		owner.playsound_local(get_turf(owner), 'sound/effects/singlebeat.ogg', 40, 1)
 
-/datum/status_effect/blooddrunk/tick() //multiply the effect of healing by 10
```

Comment:
So, this can go, because of how TG handled it--they multiply EVERYTHING but 10, then, when the buff ends, it divides everything by 10---we're just going to apply a mod that makes whatever resistances you have divide by 10, then, when the buff ends, get multiplied by 10, instead. 

Should get rid of any weirdness.